### PR TITLE
bugfix: sessions save error and some warnings from make tests

### PR DESCRIPTION
**Pull request by zeusintuivo**
Related to fix to these bugs:
[x] https://github.com/Guake/guake/pull/1775
[x] https://github.com/Guake/guake/issues/1718
[x] https://github.com/Guake/guake/issues/1737
[x] https://github.com/Guake/guake/issues/1739
[x] https://github.com/Guake/guake/issues/1753
[x] https://github.com/Guake/guake/issues/1777
[x] https://github.com/Guake/guake/issues/1779

[x] Fixes reading session.json file.

### CODE REVIEW

File: releasenotes/notes/fix_sessions_save-75bd2954cf74cf4b.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          - fr
+          - de
+
+notes_for_package_maintainers:
+  - |
+    Add notes for package maintainers here.
+
+other:
+  - |
+    Add other notes here.
```
Comment (ulidtko) [inline releasenotes/notes/fix_sessions_save-75bd2954cf74cf4b.yaml:57]: You're supposed to fill out this file and delete irrelevant parts.

**Review by Davidy22 — CHANGES_REQUESTED** [review]
There's a lot going on in here, made some comments. The slug file still needs filling in, also we have an updated CI pipeline now so you'll want to update and make sure your code passes the build. Also consider splitting this up, there's a big grab-bag of issues that you've listed that this fixes.

**Comment by Davidy22** [comment]
Coming back to this, most of the tickets that this pull request closes look like they're already closed, so this pr currently I believe is just a bunch of collisions with existing fixes and one fix for a still outstanding bug #1779. I'll give you some time to remake this pull request as just the fix to that bug, and then I'll close this one, I'll do it myself if I get around to it.

**Comment by Davidy22** [comment]
I got the time, so I did it myself. New pr with the last remaining change at #1917, closing this one.

**Pull request closed without merging**